scripts/ingest_regress_test_result.py

```diff
@@ -0,0 +1,97 @@
+#!/usr/bin/env python3
+import argparse
+import os
+import sys
+from contextlib import contextmanager
+from pathlib import Path
+
+import psycopg2
+
+CREATE_TABLE = """
+CREATE TABLE IF NOT EXISTS regress_test_results (
+    id SERIAL PRIMARY KEY,
+    reference CHAR(255),
+    revision CHAR(40),
+    build_type CHAR(16),
+    data JSONB
+)
+"""
+
+
+def err(msg):
+    print(f"error: {msg}")
+    sys.exit(1)
+
+
+@contextmanager
+def get_connection_cursor():
+    connstr = os.getenv("DATABASE_URL")
+    if not connstr:
+        err("DATABASE_URL environment variable is not set")
+    with psycopg2.connect(connstr, connect_timeout=30) as conn:
+        with conn.cursor() as cur:
+            yield cur
+
+
+def create_table(cur):
+    cur.execute(CREATE_TABLE)
+
+
+def ingest_regress_test_result(
+    cursor, reference: str, revision: str, build_type: str, data_file: Path
+):
+    values = (
+        reference,
+        revision,
+        build_type,
+        data_file.read_text(),
+    )
+    cursor.execute(
+        """
+        INSERT INTO regress_test_results (
+            reference,
+            revision,
+            build_type,
+            data
```

Comment:
What are the plans on how to work with that? Here we store plain json data without parsing into different columns. For perf test it is parsed and grafana can pick this up easily. We can write queries with manual extraction from jsonb, but it has its pros and cons

Reply:
Yes, the original idea was to collect data "as is" and visualise useful things in Grafana using Postgres JSON functions. The current implementation allows us to start to collect data and do experiments with it without worrying too much about a proper data format. If we find out that JSON won't work for our case, it'll be possible to change the DB schema and migrate the data.